include/zephyr/dt-bindings/gpio/nxp-s32-gpio.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_NXP_S32_GPIO_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_NXP_S32_GPIO_H_
+
+/**
+ * @brief NXP S32 GPIO specific flags
+ *
+ * The driver flags are encoded in the 8 upper bits of @ref gpio_dt_flags_t as
+ * follows:
+ *
+ * - Bit 8: Interrupt type (0: SIUL2 interrupt, 1: WKPU interrupt)
+ *
+ * @ingroup gpio_interface
+ * @{
+ */
+
+/** @cond INTERNAL_HIDDEN */
+#define NXP_S32_GPIO_INT_TYPE_POS	8
+#define NXP_S32_GPIO_INT_TYPE_MASK	(0x1U << NXP_S32_GPIO_INT_TYPE_POS)
+/** @endcond */
+
+/**
+ * @name NXP S32 GPIO interrupt type flags
+ * @brief NXP S32 GPIO interrupt type flags
+ * @{
+ */
+
+/** Interrupt routed to the SIUL2 External Interrupt controller */
+#define NXP_S32_GPIO_INT_EIRQ	(0x0U << NXP_S32_GPIO_INT_TYPE_POS)
```

Comment:
Hey I don't quite see a point in defining this. It's not used in the code anywhere, defining it opens the possibility of one setting both this and the `WKPU` flag which makes no sense and should be handled. I think it should just be dropped, just define`NXP_S32_GPIO_INT_WKPU` and clarify in the documentation that interrupt are routed by default through the EIRQ/SIUL2 and go to WKPU if the flag is specified.

While I'm at it: the whole EIRQ/SIUL2 naming is a fairly confusing, may not be exposed too much if you drop this one (it's odd to have SIUL2 in the documentation and EIRQ in the flag name).

Reply:
it makes sense, dropped flag as requested